common/semaphore/semaphore.go

```diff
@@ -7,7 +7,11 @@ SPDX-License-Identifier: Apache-2.0
 // Package semaphore provides an implementation of a counting semaphore.
 package semaphore
```

Comment:
We don't need a weighted semaphore so the additional dependency isn't warranted. The addition of `TryAcquire` without a context was all that was necessary here (without a context).

Reply:
Got it. Thanks. As the code for both old and new approaches present in these files, I got confused and was wondering why to add 150+ lines of code rather than importing an existing package. Now, I realized that most existing code in these files would be removed and we just need `TryAcquire` without the context. Hence, I agree that it doesn't warrant a new package dependency.